src/Tools/IdeBenchmarks/SQLitePersistentStorageBenchmark.cs

```diff
@@ -57,7 +58,8 @@ public void GlobalSetup()
 
             _storageService = new SQLitePersistentStorageService(new LocationService());
 
-            _storage = _storageService.GetStorageWorker(_workspace.CurrentSolution);
+            var solution = _workspace.CurrentSolution;
+            _storage = _storageService.GetStorageWorker(_workspace, (SolutionKey)solution, solution);
```

Comment:
the general idea now is that instead of taking a Solution/Project/Doc, there are apis taht only take what is needed by those operations to satisfy the request.  *However*, teh full snapshot is also passed along as an optional value.  That's because there are some optimizations we perform (specifically around bulk-data population) that we can do if that full snapshot is available.  However, that optimization is optional and we simply do not perform it in teh case where you're passing the fine-grained query.